plugin/trino-hive/src/test/java/io/trino/plugin/hive/metastore/thrift/TestHiveMetastoreReadAccessOperations.java

```diff
@@ -0,0 +1,340 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.metastore.thrift;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultiset;
+import io.trino.plugin.hive.HiveType;
+import io.trino.plugin.hive.TestingHivePlugin;
+import io.trino.plugin.hive.metastore.Column;
+import io.trino.plugin.hive.metastore.CountingAccessHiveMetastore;
+import io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods;
+import io.trino.plugin.hive.metastore.CountingAccessHiveMetastoreUtil;
+import io.trino.plugin.hive.metastore.Database;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastore;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import io.trino.testng.services.Flaky;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Optional;
+
+import static io.trino.connector.informationschema.InformationSchemaMetadata.MAX_PREFIXES_COUNT;
+import static io.trino.plugin.hive.HiveStorageFormat.ORC;
+import static io.trino.plugin.hive.TableType.MANAGED_TABLE;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_ALL_DATABASES;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_ALL_TABLES_WITH_PARAMETER;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_ALL_VIEWS_WITH_PARAMETER;
+import static io.trino.plugin.hive.metastore.CountingAccessHiveMetastore.Methods.GET_TABLE;
+import static io.trino.plugin.hive.metastore.PrincipalPrivileges.NO_PRIVILEGES;
+import static io.trino.plugin.hive.metastore.StorageFormat.fromHiveStorageFormat;
+import static io.trino.plugin.hive.metastore.file.FileHiveMetastore.createTestingFileHiveMetastore;
+import static io.trino.testing.TestingSession.testSessionBuilder;
+
+@Test(singleThreaded = true) // metastore invocation counters shares mutable state so can't be run from many threads simultaneously
```

Comment:
Please copy-paste tests but for `system.jdbc`. To capture how things looks today and prevent any regression. Please do not fix any encountered problem as part of this PR.

Reply:
Done but I feel we should use `trino-jdbc` to have exact queries used by JDBC

Reply:
`system` is also a plain connector so the fact it is used in jdbc is not that much important. We need are testing a system connector integration with hive connector here. Putting JDBC into the same test sounds like an overkill as it is already a bit complex.

Reply:
we already exercise `system.jdbc` directly in other tests (see eg BCT).
better to do that directly than not do that at all.

testing via JDBC could be more verbose, i wouldn't insist on that, but wouldn't also oppose strongly.

Reply:
> `system` is also a plain connector so the fact it is used in jdbc is not that much important.

i don't agree.
`system.jdbc` is supposed to be accessed via JDBC and we care primarily about usage patterns generated by the JDBC driver.
at this point, i wouldn't bother about performance of, say, joins or union queries or anything like that on top of ``system.jdbc``, because it's outside of typical usage pattern